src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs

```diff
@@ -236,7 +236,7 @@ public DocumentStateChecksums(params object[] children) : base(WellKnownSynchron
 
             if (searchingChecksumsLeft.Remove(Text))
             {
-                result[Text] = await state.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                result[Text] = await SerializableSourceText.FromTextDocumentStateAsync(state, cancellationToken).ConfigureAwait(false);
```

Comment:
📝 This pull request's _sole purpose_ is to avoid the use of `GetTextAsync` here, and store recoverable text in `SolutionAsset` instead of `SourceText`.